### PR TITLE
Develop guide

### PR DESCRIPTION
**Pull request by fermenreq**

### CODE REVIEW

File: doc/connection-to-the-internet-of-things/introduction.md [hunk 1]
```diff
@@ -24,47 +24,24 @@ it is depicted in the following picture: 
 
 **FIWARE IoT Device Management GE architecture**
 
-This simple scenario considers only two Generic Enablers
+This simple scenario has got two basics
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/introduction.md]: This simple scenarios relies on two different components:

File: doc/connection-to-the-internet-of-things/introduction.md [hunk 2]
```diff
@@ -24,47 +24,24 @@ it is depicted in the following picture: 
...
 
 -   **Orion ContextBroker:** That remains as the main front-end for
     developers. Developers access IoT data as attributes of entities
     representing devices and Developers may also send commands to
     devices by updating command-related attributes, providing they have
     access rights for that operation.
--   **IDAS Backend Device Manager:** This component stays at the
-    southbound of the Orion ContextBroker and it is used by IoT
-    integrators to connect devices in this scenario.
+    You can see the lastest version [here](https://github.com/Fiware/context.Orion/tree/release/1.7.0)
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/introduction.md]: Please do not reference concrete versions, as they will be outdated soon

File: doc/connection-to-the-internet-of-things/introduction.md [hunk 3]
```diff
@@ -24,47 +24,24 @@ it is depicted in the following picture: 
...
+-   **[IoT Agents](https://github.com/Fiware?utf8=%E2%9C%93&q=IoTAgent):** These components stay at the southbound of the 
+    Orion ContextBroker and they are used by IoT integrators to connect devices in this scenario. IoT Agents support several
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/introduction.md]: the style of these lines is strange, are you using tabs or similar?

File: doc/connection-to-the-internet-of-things/introduction.md [hunk 4]
```diff
@@ -24,47 +24,24 @@ it is depicted in the following picture: 
...
-    [here](https://github.com/telefonicaid/lightweightm2m-iotagent).
-    FIWARE Lab public instance details are
-    [here](http://catalogue.fiware.org/enablers/backend-device-management-idas/instances).
-
-For simplicity, in this article Ultralight2.0/HTTP is explained. Please
-refer to the FIWARE Catalogue to find out more information on the other
-options.  
- If you are interested to connect your application to the Internet of
-Things, keep on reading:
+For simplicity, in this article Ultralight2.0/HTTP is explained. To find out more information plese visit Github site.
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/introduction.md]: Ultralight2.0/HTTP is used?

Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/introduction.md]: please link to Github properly

File: doc/development-context-aware-applications/deep-dive.md [hunk 5]
```diff
@@ -1,12 +1,5 @@
 Should you wish to learn much more about this, please check: 
 
    - [Introduction to NGSI API and Orion Context Broker](http://bit.ly/fiware-orion)
-   - [Github repository (reference implementation) ](https://github.com/Fiware/context.Orion)
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/deep-dive.md]: Please keep the link to the Github repository as we agreed

File: doc/development-context-aware-applications/introduction.md [hunk 6]
```diff
@@ -44,12 +44,10 @@ a context-aware application check out:
     - [How to update and query context information](v2/how-to-update-and-query-context-information.md)
     - [How to subscribe to changes on context information](v2/how-to-subscribe-to-changes-on-context-information.md)
     - [How to perform geo-location queries](v2/how-to-perform-geo-located-queries.md)
+    
+  By other hand, you can see a the full API [here](http://fiware.github.io/specifications/ngsiv2/latest/cookbook/)
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/introduction.md]: On the other hand, you can browse the full API description [here](http://fiware.github.io/specifications/ngsiv2/latest/cookbook/)

File: doc/development-context-aware-applications/introduction.md [hunk 7]
```diff
@@ -44,12 +44,10 @@ a context-aware application check out:
...
 
-- Using NGSI v1: 
-    - [How to update and query context information](how-to-update-and-query-context-information.md)
-    - [How to connect to external context providers](how-to-connect-to-external-context-providers.md)
-    - [How to subscribe to changes on context information](how-to-subscribe-to-changes-on-context-information.md)
-    - [How to perform geo-location queries](how-to-perform-geo-located-queries.md)
+It exists an early NGSI version, call NGSI v1 but it has been largely overpassed by the current NGSIv2 version and it will be deprecated soon. Anyway you can check it [here](http://fiware.github.io/context.Orion/api/v1/)
```
Comment (jmcanterafonseca) [inline doc/development-context-aware-applications/introduction.md]: There is an early FIWARE NGSI HTTP binding, named NGSI v1, but ...

File: doc/connection-to-the-internet-of-things/deep-dive.md [hunk 8]
```diff
@@ -1,13 +1,11 @@
-Should you wish to learn much more about this, please check: 
+Should you wish to learn much more about this. Currently exists the following IoT Agents:
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/deep-dive.md]: Should you wish to learn much more about this, please check the following IoT Agents:

File: doc/connection-to-the-internet-of-things/deep-dive.md [hunk 9]
```diff
@@ -1,13 +1,11 @@
...
-   - [Latest stable release - IoT Agent UL 2.0](https://github.com/Fiware/iot.IoTagent-UL/releases/latest)
-   
-Other resources (advanced):
-   
-   - [IoT Agent JSON](https://github.com/Fiware/iot.IoTagent-JSON)
-   - [IoT Agent LWM2M](https://github.com/Fiware/iot.IoTagent-LWM2M)
-   - [IoT Agent Node.js Library - Framework to develop new IoT Agents](https://github.com/Fiware/iot.IoTagent-node-lib)
\ No newline at end of file
+- [IoT Agent Ultralight2.0](https://github.com/Fiware/iot.IoTagent-UL): This IoT Agent is designed to be a bridge between the UltraLight2.0 protocol and NGSI. FIWARE Lab public instance details are [here](https://catalogue.fiware.org/enablers/backend-device-management-idas/instances). 
+- [IoT LWM2M](https://github.com/Fiware/iot.IoTagent-LWM2M): This IoT Agent is designed to be a bridge between an Lightweight M2M protocol and an Orion Context Broker..
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/deep-dive.md]: between the

File: doc/connection-to-the-internet-of-things/deep-dive.md [hunk 10]
```diff
@@ -1,13 +1,11 @@
...
-   
-Other resources (advanced):
-   
-   - [IoT Agent JSON](https://github.com/Fiware/iot.IoTagent-JSON)
-   - [IoT Agent LWM2M](https://github.com/Fiware/iot.IoTagent-LWM2M)
-   - [IoT Agent Node.js Library - Framework to develop new IoT Agents](https://github.com/Fiware/iot.IoTagent-node-lib)
\ No newline at end of file
+- [IoT Agent Ultralight2.0](https://github.com/Fiware/iot.IoTagent-UL): This IoT Agent is designed to be a bridge between the UltraLight2.0 protocol and NGSI. FIWARE Lab public instance details are [here](https://catalogue.fiware.org/enablers/backend-device-management-idas/instances). 
+- [IoT LWM2M](https://github.com/Fiware/iot.IoTagent-LWM2M): This IoT Agent is designed to be a bridge between an Lightweight M2M protocol and an Orion Context Broker.. 
+- [IoT Agent JSON](https://github.com/Fiware/iot.IoTagent-JSON): This agent is designed to be a bridge between an MQTT+JSON based protocol and an Orion Context Broker.
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/deep-dive.md]: MQTT+JSON protocol and NGSI

File: doc/connection-to-the-internet-of-things/deep-dive.md [hunk 11]
```diff
@@ -1,13 +1,11 @@
...
+- [IoT Agent Node Lib](https://github.com/Fiware/iot.IoTagent-node-lib): library to create new agents. This core library allows to develop new agents for specific southbound protocols/standards/messages.
+
+There are, already, [IoT Agents instances up and running](https://catalogue.fiware.org/enablers/backend-device-management-idas/instances).These would be useful for testing purpose and learning about how to interact between IoT Agents and Orion Context Broker.
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/deep-dive.md]: There are existing .... available

File: doc/connection-to-the-internet-of-things/introduction.md [hunk 12]
```diff
@@ -17,54 +17,28 @@ it is tricky to find a global deployment.  
...
     representing devices and Developers may also send commands to
     devices by updating command-related attributes, providing they have
     access rights for that operation.
--   **IDAS Backend Device Manager:** This component stays at the
-    southbound of the Orion ContextBroker and it is used by IoT
-    integrators to connect devices in this scenario.
+    You can see the lastest version [here](https://github.com/Fiware/context.Orion)
+-   **[IoT Agents](https://github.com/Fiware?utf8=%E2%9C%93&q=IoTAgent):** These components stay at the southbound of the 
+    Orion ContextBroker and they are used by IoT integrators to connect devices in this scenario. IoT Agents support several 
+    IoT protocols with a modular architecture Therefore, integrators need to determine first which protocol they will be using
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/introduction.md]: with a modular architecture. Therefore,

File: doc/connection-to-the-internet-of-things/deep-dive.md [hunk 13]
```diff
@@ -1,13 +1,12 @@
...
\ No newline at end of file
+- [IoT Agent Ultralight2.0](https://github.com/Fiware/iot.IoTagent-UL): This IoT Agent is designed to be a bridge between the UltraLight2.0 protocol and NGSI. FIWARE Lab public instance details are [here](https://catalogue.fiware.org/enablers/backend-device-management-idas/instances). 
+- [IoT LWM2M](https://github.com/Fiware/iot.IoTagent-LWM2M): This IoT Agent is designed to be a bridge between the Lightweight M2M protocol and an Orion Context Broker.. 
+- [IoT Agent JSON](https://github.com/Fiware/iot.IoTagent-JSON): This agent is designed to be a bridge between the MQTT+JSON protocol 
+and NGSI.
+- [IoT Agent Node Lib](https://github.com/Fiware/iot.IoTagent-node-lib): library to create new agents. This core library allows to develop new agents for specific southbound protocols/standards/messages.
+
+There are existing [IoT Agents instances up and running](https://catalogue.fiware.org/enablers/backend-device-management-idas/instances) available.These would be useful for testing purpose and learning about how to interact between IoT Agents and Orion Context Broker.
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/deep-dive.md]: space missing between the '.' and These

File: doc/connection-to-the-internet-of-things/deep-dive.md [hunk 14]
```diff
@@ -1,13 +1,12 @@
...
-   - [Latest stable release - IoT Agent UL 2.0](https://github.com/Fiware/iot.IoTagent-UL/releases/latest)
-   
-Other resources (advanced):
-   
-   - [IoT Agent JSON](https://github.com/Fiware/iot.IoTagent-JSON)
-   - [IoT Agent LWM2M](https://github.com/Fiware/iot.IoTagent-LWM2M)
-   - [IoT Agent Node.js Library - Framework to develop new IoT Agents](https://github.com/Fiware/iot.IoTagent-node-lib)
\ No newline at end of file
+- [IoT Agent Ultralight2.0](https://github.com/Fiware/iot.IoTagent-UL): This IoT Agent is designed to be a bridge between the UltraLight2.0 protocol and NGSI. FIWARE Lab public instance details are [here](https://catalogue.fiware.org/enablers/backend-device-management-idas/instances). 
+- [IoT LWM2M](https://github.com/Fiware/iot.IoTagent-LWM2M): This IoT Agent is designed to be a bridge between the Lightweight M2M protocol and an Orion Context Broker..
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/deep-dive.md]: and NGSI.

File: doc/connection-to-the-internet-of-things/deep-dive.md [hunk 15]
```diff
@@ -1,13 +1,12 @@
...
-   
-   - [IoT Agent JSON](https://github.com/Fiware/iot.IoTagent-JSON)
-   - [IoT Agent LWM2M](https://github.com/Fiware/iot.IoTagent-LWM2M)
-   - [IoT Agent Node.js Library - Framework to develop new IoT Agents](https://github.com/Fiware/iot.IoTagent-node-lib)
\ No newline at end of file
+- [IoT Agent Ultralight2.0](https://github.com/Fiware/iot.IoTagent-UL): This IoT Agent is designed to be a bridge between the UltraLight2.0 protocol and NGSI. FIWARE Lab public instance details are [here](https://catalogue.fiware.org/enablers/backend-device-management-idas/instances). 
+- [IoT LWM2M](https://github.com/Fiware/iot.IoTagent-LWM2M): This IoT Agent is designed to be a bridge between the Lightweight M2M protocol and an Orion Context Broker.. 
+- [IoT Agent JSON](https://github.com/Fiware/iot.IoTagent-JSON): This agent is designed to be a bridge between the MQTT+JSON protocol 
+and NGSI.
+- [IoT Agent Node Lib](https://github.com/Fiware/iot.IoTagent-node-lib): library to create new agents. This core library allows to develop new agents for specific southbound protocols/standards/messages.
```
Comment (jmcanterafonseca) [inline doc/connection-to-the-internet-of-things/deep-dive.md]: I would only add 'protocols'

File: doc/development-context-aware-applications/deep-dive.md (hunk 5 above)
Comment (fermenreq) [reply doc/development-context-aware-applications/deep-dive.md]: Thanks, just added

**Comment by jmcanterafonseca** [comment]
LGTM

**Pull request merged**